### PR TITLE
feat: Updated chip background color based on tier level

### PR DESCRIPTION
**Pull request by VaishaliDesai**
Hello,

First of all, I would like to thank to the owner and reviewers of rtchat to let me contribute to this amazing flutter project. This is my first ever flutter PR so I would appreciate all your suggestions and guidance to improve.

This PR updates the background color of chip based on its tier level in chat history. 

Issue - https://github.com/muxable/rtchat/issues/929

Below screenshot shows new colors based on the tier levels -
Screenshot 2023-02-19 at 23 53 42
Screenshot 2023-02-19 at 23 47 47
Screenshot 2023-02-19 at 23 49 34

### CODE REVIEW

File: lib/components/chat_history/twitch/subscription_event.dart [hunk 1]
```diff
@@ -72,6 +72,16 @@ class TwitchSubscriptionMessageEventWidget extends StatelessWidget {
   const TwitchSubscriptionMessageEventWidget(this.model, {Key? key})
       : super(key: key);
 
+  Color chipBackgroundColor(String tier, BuildContext context) {
+    if (tier == '2000') {
+      return  const Color(0xFFFFEA00);
```
Comment (kevmo314) [inline lib/components/chat_history/twitch/subscription_event.dart:77]:
```suggestion
      return const Color(0xFFFFEA00);
```

File: lib/components/chat_history/twitch/subscription_event.dart [hunk 2]
```diff
@@ -72,6 +72,16 @@ class TwitchSubscriptionMessageEventWidget extends StatelessWidget {
...
+    } else if (tier == '3000') {
+      return  const Color(0xFFD2D2e6);
```
Comment (kevmo314) [inline lib/components/chat_history/twitch/subscription_event.dart:79]:
```suggestion
      return const Color(0xFFD2D2E6);
```

File: lib/components/chat_history/twitch/subscription_event.dart [hunk 3]
```diff
@@ -72,6 +72,16 @@ class TwitchSubscriptionMessageEventWidget extends StatelessWidget {
   const TwitchSubscriptionMessageEventWidget(this.model, {Key? key})
       : super(key: key);
 
+  Color chipBackgroundColor(String tier, BuildContext context) {
```
Comment (kevmo314) [inline lib/components/chat_history/twitch/subscription_event.dart:75]: Switch the order of the arguments, `BuildContext` should typically come first.

**Review by kevmo314 — CHANGES_REQUESTED** [review]
Nice work! This looks good, just a couple minor changes and we'll get this released :)

File: lib/components/chat_history/twitch/subscription_event.dart (hunk 1 above)
Comment (SputNikPlop) [inline lib/components/chat_history/twitch/subscription_event.dart:77]:
```suggestion
      return const Color(0xFFD2D2E6);
```
Tier 2 is not commonly used lets change it to the lighter color.

File: lib/components/chat_history/twitch/subscription_event.dart (hunk 2 above)
Comment (SputNikPlop) [inline lib/components/chat_history/twitch/subscription_event.dart:79]:
```suggestion
      return const Color(0xFFFFEA00);
```
Tier 3 is more commonly subscribed to lets make it a distinguished color.

**Review by SputNikPlop — CHANGES_REQUESTED** [review]

File: lib/components/chat_history/twitch/subscription_event.dart [hunk 4]
```diff
@@ -72,6 +72,16 @@ class TwitchSubscriptionMessageEventWidget extends StatelessWidget {
   const TwitchSubscriptionMessageEventWidget(this.model, {Key? key})
       : super(key: key);
 
+  Color chipBackgroundColor(BuildContext context, String tier) {
+    if (tier == '2000') {
+      return const Color(0xFFD2D2e6);
```
Comment (SputNikPlop) [inline lib/components/chat_history/twitch/subscription_event.dart:77]:
```suggestion
      return const Color(0xFFD2D2E6);
```

**Review by SputNikPlop — CHANGES_REQUESTED** [review]

**Review by SputNikPlop — APPROVED** [review]
LGTM

**Review by kevmo314 — APPROVED** [review]

**Comment by VaishaliDesai** [comment]
Thank you both @kevmo314 and @SputNikPlop for your reviews, the CI seems to be failing. Not sure why!

**Comment by kevmo314** [comment]
> Thank you both @kevmo314 and @SputNikPlop for your reviews, the CI seems to be failing. Not sure why!

Oh that's Codeball, that check is fine. It seems to fail periodically but is only for reviewing. I've merged the PR for you, thanks again for the contribution!

**Pull request merged**